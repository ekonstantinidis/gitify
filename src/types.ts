export enum Appearance {
  SYSTEM = 'SYSTEM',
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}

export type RadioGroupItem = {
  label: string;
  value: string;
};
