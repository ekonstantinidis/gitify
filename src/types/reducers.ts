import { Notification } from './github';

export interface EnterpriseAccount {
  hostname: string;
  token: string;
}

export interface AppState {
  auth: AuthState;
  notifications: NotificationsState;
  settings: SettingsState;
}

export interface AuthState {
  response: {};
  token?: string;
  isFetching: boolean;
  failed: boolean;
  enterpriseAccounts: EnterpriseAccount[];
}

export interface AccountNotifications {
  hostname: string;
  notifications: Notification[];
}

export interface NotificationsState {
  response: AccountNotifications[];
  isFetching: boolean;
  failed: boolean;
}

export interface SettingsState {
  participating: boolean;
  playSound: boolean;
  showNotifications: boolean;
  markOnClick: boolean;
  openAtStartup: boolean;
  showDockIcon: boolean;
}
