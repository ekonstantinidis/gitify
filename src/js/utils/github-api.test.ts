import { formatReason, getNotificationTypeIcon } from './github-api';
import { Reason, SubjectType } from '../../types/github';

describe('./utils/github-api.ts', () => {
  it('should format the notification reason', () => {
    expect(formatReason('assign')).toMatchSnapshot();
    expect(formatReason('author')).toMatchSnapshot();
    expect(formatReason('comment')).toMatchSnapshot();
    expect(formatReason('invitation')).toMatchSnapshot();
    expect(formatReason('manual')).toMatchSnapshot();
    expect(formatReason('mention')).toMatchSnapshot();
    expect(formatReason('review_requested')).toMatchSnapshot();
    expect(formatReason('security_alert')).toMatchSnapshot();
    expect(formatReason('state_change')).toMatchSnapshot();
    expect(formatReason('subscribed')).toMatchSnapshot();
    expect(formatReason('team_mention')).toMatchSnapshot();
    expect(formatReason('ci_activity')).toMatchSnapshot();
    expect(formatReason('something_else_unknown' as Reason)).toMatchSnapshot();
  });

  it('should get the notification type icon', () => {
    expect(getNotificationTypeIcon('Issue')).toBe('issue-opened');
    expect(getNotificationTypeIcon('PullRequest')).toBe('git-pull-request');
    expect(getNotificationTypeIcon('Commit')).toBe('git-commit');
    expect(getNotificationTypeIcon('Release')).toBe('tag');
    expect(getNotificationTypeIcon('RepositoryVulnerabilityAlert')).toBe(
      'alert'
    );
    expect(getNotificationTypeIcon('CheckSuite')).toBe('sync');
    expect(getNotificationTypeIcon('Unknown' as SubjectType)).toBe('question');
  });
});
