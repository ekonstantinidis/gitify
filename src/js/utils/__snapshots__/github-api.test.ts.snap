// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./utils/github-api.ts should format the notification reason 1`] = `
Object {
  "description": "You were assigned to the issue.",
  "type": "Assign",
}
`;

exports[`./utils/github-api.ts should format the notification reason 2`] = `
Object {
  "description": "You created the thread.",
  "type": "Author",
}
`;

exports[`./utils/github-api.ts should format the notification reason 3`] = `
Object {
  "description": "You commented on the thread.",
  "type": "Comment",
}
`;

exports[`./utils/github-api.ts should format the notification reason 4`] = `
Object {
  "description": "You accepted an invitation to contribute to the repository.",
  "type": "Invitation",
}
`;

exports[`./utils/github-api.ts should format the notification reason 5`] = `
Object {
  "description": "You subscribed to the thread (via an issue or pull request).",
  "type": "Manual",
}
`;

exports[`./utils/github-api.ts should format the notification reason 6`] = `
Object {
  "description": "You were specifically @mentioned in the content.",
  "type": "Mention",
}
`;

exports[`./utils/github-api.ts should format the notification reason 7`] = `
Object {
  "description": "You, or a team you're a member of, were requested to review a pull request.",
  "type": "Review Requested",
}
`;

exports[`./utils/github-api.ts should format the notification reason 8`] = `
Object {
  "description": "GitHub discovered a security vulnerability in your repository.",
  "type": "Security Alert",
}
`;

exports[`./utils/github-api.ts should format the notification reason 9`] = `
Object {
  "description": "You changed the thread state (for example, closing an issue or merging a pull request).",
  "type": "State Change",
}
`;

exports[`./utils/github-api.ts should format the notification reason 10`] = `
Object {
  "description": "You're watching the repository.",
  "type": "Subscribed",
}
`;

exports[`./utils/github-api.ts should format the notification reason 11`] = `
Object {
  "description": "You were on a team that was mentioned.",
  "type": "Team Mention",
}
`;

exports[`./utils/github-api.ts should format the notification reason 12`] = `
Object {
  "description": undefined,
  "type": "Workflow Run",
}
`;

exports[`./utils/github-api.ts should format the notification reason 13`] = `
Object {
  "description": "The reason for this notification is not supported by the app.",
  "type": "Unknown",
}
`;
